minidump-processor/src/op_analysis.rs

```diff
@@ -330,46 +622,116 @@ mod amd64 {
             Ok(())
         }
 
-        /// Indirect memory accesses include instructions which change the instruction pointer.
-        fn indirect(
+        fn implicit_accesses(
             &self,
-            accesses: &mut Vec<MemoryAccess>,
-            decoded_instruction: &Instruction,
+            accesses: &mut Vec<MemoryAddressInfo>,
+            decoded_instruction: Instruction,
         ) -> Result<(), OpAnalysisError> {
-            let mut push_indirect_access = |address| {
-                accesses.push(MemoryAccess {
+            let mut push_implicit_access = |address, access_type| {
+                accesses.push(MemoryAddressInfo {
                     address,
                     is_likely_null_pointer_dereference: address == 0,
                     is_likely_guard_page: false,
-                    size: Some(1),
+                    size: Some(1), // TODO: correct size?
+                    access_type,
                 });
             };
 
+            let Some(opcode) = CommonOpcode::from_amd64_opcode(decoded_instruction.opcode()) else {
+                return Ok(());
+            };
+            match opcode {
+                CommonOpcode::CALL | CommonOpcode::PUSH => {
+                    if let Ok(rsp) = self.context.get_regspec(RegSpec::rsp()) {
+                        push_implicit_access(rsp, Some(OperandAccessType::Write));
+                    }
+                }
+                CommonOpcode::IRET
+                | CommonOpcode::IRETD
+                | CommonOpcode::IRETQ
+                | CommonOpcode::POP
+                | CommonOpcode::RETF
+                | CommonOpcode::RETURN => {
+                    if let Ok(rsp) = self.context.get_regspec(RegSpec::rsp()) {
+                        push_implicit_access(rsp, Some(OperandAccessType::Read));
+                    }
+                }
+                CommonOpcode::ADD
+                | CommonOpcode::CMP
+                | CommonOpcode::DEC
+                | CommonOpcode::INC
+                | CommonOpcode::JMP
+                | CommonOpcode::JMPF
+                | CommonOpcode::MOV
+                | CommonOpcode::MOVAPS
+                | CommonOpcode::MOVUPS
+                | CommonOpcode::SUB => (),
+            }
+            Ok(())
+        }
+    }
+
+    impl<'a> GetInstructionPointerUpdate<'a> {
```

Comment:
This is another example of a verb masquerading as a noun. Perhaps it makes more sense to have this:

```Rust
struct InstructionPointerUpdate {
    address_info: MemoryAddressInfo,
}

impl InstructionPointerUpdate {
    pub fn from_instruction(
        context: &MinidumpContext,
        memory_list: Option<&minidump::UnifiedMemoryList<'_>>,
        stack_memory: Option<minidump::UnifiedMemory<'_, '_>>,
        decoded_instruction: Instruction,
    ) -> Result<Option<InstructionPointerUpdate>, OpAnalysisError> {
    }
}
```